src/main/kotlin/step2/parser/NodeParserImpl.kt

```diff
@@ -0,0 +1,36 @@
+package step2.parser
+
+import step2.calculator.Node
+import java.util.regex.Pattern
+
+const val PLUS = "+"
+const val MINUS = "-"
+const val MULTIPLY = "*"
+const val DIVIDE = "/"
+
+class NodeParserImpl : NodeParser {
+    private val regexForNode = Regex("((?<=([-*/+\\s]))|(?=([-*/+\\s])))")
+    private val patternForNumber = Pattern.compile("^[0-9]+$")
+    override fun parse(text: String): List<Node> {
+        return text
+            .split(regexForNode)
+            .trimAll()
+            .exceptEmpty()
+            .convertAllToNode()
+    }
+
+    fun List<String>.trimAll() = map { it.trim() }
+
+    fun List<String>.exceptEmpty() = filter { it != "" }
```

Comment:
`String`에 대해 호출할 수 있는 `isNotEmpty()` 또는 `isNotBlank()`가 있습니다.

```suggestion
    fun List<String>.exceptEmpty() = filter { it.isNotBlank() }
```